### PR TITLE
More release note entries

### PR DESCRIPTION
**Pull request by nojaf**
## Description

This adds some more release note entries of recent changes.
I didn't do any PR that was related to VS, as I don't feel confident enough to write those.

@dawedawe could you try out my instructions to run the docs website locally, please.

## Checklist

- [ ] Test cases added
- [ ] Performance benchmarks added in case of performance changes
- [X] Release notes entry updated:
    > Please make sure to add an entry with short succinct description of the change as well as link to this pull request to the respective release notes file, if applicable.
    >
    > Release notes files:
    > - If anything under `src/Compiler` has been changed, please make sure to make an entry in `docs/release-notes/.FSharp.Compiler.Service/<version>.md`, where `<version>` is usually "highest" one, e.g. `42.8.200`
    > - If language feature was added (i.e. `LanguageFeatures.fsi` was changed), please add it to `docs/releae-notes/.Language/preview.md`
    > - If a change to `FSharp.Core` was made, please make sure to edit `docs/release-notes/.FSharp.Core/<version>.md` where version is "highest" one, e.g. `8.0.200`.

    > Information about the release notes entries format can be found in the [documentation](https://fsharp.github.io/fsharp-compiler-docs/release-notes/About.html).
    > Example:
    > * More inlines for Result module. ([PR #16106](https://github.com/dotnet/fsharp/pull/16106))
    > * Correctly handle assembly imports with public key token of 0 length. ([Issue #16359](https://github.com/dotnet/fsharp/issues/16359), [PR #16363](https://github.com/dotnet/fsharp/pull/16363))
    > *`while!` ([Language suggestion #1038](https://github.com/fsharp/fslang-suggestions/issues/1038), [PR #14238](https://github.com/dotnet/fsharp/pull/14238))

    > **If you believe that release notes are not necessary for this PR, please add `NO_RELEASE_NOTES` label to the pull request.**

### CODE REVIEW

**Comment by nojaf** [comment]
I didn't add any entry for https://github.com/dotnet/fsharp/pull/16393, I'm not sure how to describe that one properly.

**Comment by vzarytovskii** [comment]
While you at it, can I please ask you to change PR template as well, describing that we're using specific format?

**Comment by vzarytovskii** [comment]
Ah, nvm, there's a link, I'm blind.

File: docs/running-documentation-locally.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+index: 999
+---
+# Running the documentation locally
+
+The source of this documentation website is hosted on https://github.com/fsharp/fsharp-compiler-docs.  
+
+## Setup
+
+`fsharp/fsharp-compiler-docs` will clone the `dotnet/fsharp` repository first to generate the documentation.  
+You can however, easily run the documentation locally and modify the `docs` from `dotnet/fsharp`.
```
Comment (psfinaki) [inline docs/running-documentation-locally.md:14]: Maybe worth adding a sentence on why someone might need this.

Comment (nojaf) [reply docs/running-documentation-locally.md:14]: Isn't that a little obvious? You would like to locally how your changes will reflect the documentation website.

**Comment by dawedawe** [comment]
@nojaf the instructions work 👍

File: docs/running-documentation-locally.md (hunk 1 above)
Comment (psfinaki) [reply docs/running-documentation-locally.md:14]: 👀 

Well personally I would be probably motivated to do this only when doing big changes on that front. 
But as you wish, feel free to keep it as it is if you will.

**Review by psfinaki — APPROVED** [review]

**Review by T-Gro — APPROVED** [review]

**Pull request merged**